server/rooms.js

```diff
@@ -1693,8 +1693,12 @@ let Rooms = Object.assign(getRoom, {
 				user.inviteOnlyNextBattle = false;
 			}
 		}
-		if (options.tour && !room.tour.modjoin) inviteOnly = [];
-		if (inviteOnly.length) {
+		const prefix = /** @type {RoomBattle} */(room.game).forcedPublic();
+		if (prefix && inviteOnly.length) {
```

Comment:
This should probably only get checked in rated battles, as long as I'm understanding how ladder tours work properly.

Reply:
Also, doesn't this totally break setting invite only for ionext + no prefix?

Reply:
Added a `room.rated` check. `/ionext` without a prefix should be handled in the `else if`